### ISSUE TITLE
Boolean literals are passed as integers from macros to other macros/templates

### ISSUE THREAD

**Issue by endragor** [nim-lang/Nim#10719]
Nim ec6e5681daddde1ff81f9235a6f5348d46ab9eff

It seems that passing `true`/`false` literals within macros gets them converted to integer literals internally, which further results in errors if you try to use them as booleans. This worked as expected in 0.18.0.

### Example
```nim
import macros

macro someMacro*(): typed =
  template tmpl(boolean) =
    when boolean:
      echo "it's true!"
  result = getAst(tmpl(true))

someMacro()
```

### Current Output
```
test.nim(9, 10) template/generic instantiation of `someMacro` from here
test.nim(7, 23) Error: type mismatch: got <int literal(1)> but expected 'bool'
```

**Comment by bluenote10** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/7375?

**Comment by andreaferretti** [comment]
Seems a duplicate of https://github.com/nim-lang/Nim/issues/7375

**Comment by krux02** [comment]
it is certainly related.

**Comment by narimiran** [comment]
Closed as a duplicate of #7375.

@endragor if you want, post your example in a comment there.

**Comment by krux02** [comment]
I did some investigations now, I don't think it's a duplicate, but it is very much related. But keep this issue close I will merge the issues manually.
